src/web/releases.rs

```diff
@@ -497,6 +497,69 @@ impl Default for Search {
     }
 }
 
+fn redirect_to_random_crate(req: &Request, conn: &mut PoolClient) -> IronResult<Response> {
+    // since there is a chance of this query returning an empty result,
+    // we retry a certain amount of times, and log a warning.
+    for _ in 0..20 {
+        let rows = ctry!(
+            req,
+            conn.query(
+                "WITH params AS (
+                    -- get maximum possible id-value in crates-table
+                    SELECT last_value AS max_id FROM crates_id_seq
+                )
+                SELECT
+                    crates.name,
+                    releases.version,
+                    releases.target_name
+                FROM (
+                    -- generate 500 random numbers in the ID-range. 
+                    -- this might have to be increased when we have to repeat 
+                    -- this query too often. 
+                    -- it depends on the percentage of crates with > 100 stars
+                    SELECT DISTINCT 1 + trunc(random() * params.max_id)::INTEGER AS id
+                    FROM params, generate_series(1, 500)
```

Comment:
Hard-coding 500 makes me a little nervous. Could we generate this live based on production info? Or does that slow down the query too much?

Reply:
generating that value live slows down the query too much, I valued speed higher.

Reply:
the number depends on the ratio of crates >100 stars

Reply:
while thinking about this, I have another idea (but not likely it's faster)

Reply:
Let me know if it works out :)

Reply:
didn't help 😢 was too slow :)